sdks/python/apache_beam/runners/worker/statesampler.pyx

```diff
@@ -33,6 +33,11 @@ transition between these states of different Operations.  Each time the sampling
 thread queries the current state, the time spent since the previous sample is
 attributed to that state and accumulated.  Over time, this allows a granular
 runtime profile to be produced.
+
+Also measures time spent for the current active transition. For shorter
+transitions (comparable to the sampling period), this may not be
+accurate since there aren't enough samples, but provides a good estimate for
+longer transitions.
```

Comment:
<!--new_thread; commit:c0086ba69497a5cbd3269f1d691831ca78563dcc; resolved:0-->
Can you provide more extensive documentation on how the active and completed values are used?  As is, it is unclear what parts of this code are essential to sampling, and which are used for counting active time.